src/framework/ui/modal/modal.component.tsx

```diff
@@ -107,7 +107,11 @@ export class Modal extends React.Component<ModalProps> {
   private id: string = '';
 
   public componentWillReceiveProps(nextProps: ModalProps): void {
```

Comment:
I think that this is not a good place to perform this operation
https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

Should you prefer [componentDidUpdate](https://reactjs.org/docs/react-component.html#componentdidupdate)?

Reply:
Will think about it.